src/Layout/with-layout.js

```diff
@@ -0,0 +1,209 @@
+// @flow weak
+import React, { PropTypes } from 'react';
+import { createStyleSheet } from 'jss-theme-reactor';
+import classNames from 'classnames';
+import layoutStyles from './layout-styles';
+
+export const styleSheet = createStyleSheet('Block', layoutStyles);
+
+const BreakPointEnum = PropTypes.oneOf(['xl', 'lg', 'md', 'sm', 'xs']);
+
+/*
+* Make sure that the `layout` property is included before including properties
+*
+* - padding
+* - margin
+* - fill
+* - align
+* - scroll
+*/
+const layoutExists = (props, propName, componentName) => {
+  if (props[propName] && !props.layout) {
+    return new Error(
+      `\`layout\` property must be specified for \`${componentName}\` before using \`${propName}\` property`
+    );
+  }
+  return null;
+};
+/*
+* Make sure that the `flex` property is included before including properties
+*
+* - order
+* - offset
+* - wrap
+*/
+const flexExists = (props, propName, componentName) => {
+  if (props[propName] && !props.flex) {
+    return Error(
+      `\`flex\` property must be specified for \`${componentName}\` before using \`${propName}\` property`
+    );
+  }
+  return null;
+};
+const FlexProps = {
+  layout: PropTypes.oneOf(['row', 'column']),
+  padding: PropTypes.oneOfType([
+    layoutExists,
+    PropTypes.bool,
+  ]),
+  margin: PropTypes.oneOfType([
+    layoutExists,
+    PropTypes.bool,
+  ]),
+  fill: PropTypes.oneOfType([
+    layoutExists,
+    PropTypes.bool,
+  ]),
+  align: PropTypes.oneOf([
+    'start',
+    'center',
+    'end',
+    'stretch',
+    'space-between',
```

Comment:
I was looking into the implementation for a custom layout component at the office.
Looks like `space-between` and `space-around` shouldn't be here.